provider/pkg/provider/remote/copy.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
+
+import (
+	"errors"
+
+	"github.com/pulumi/pulumi-go-provider/infer"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/archive"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/asset"
+)
+
+type Copy struct{}
+
+var _ = (infer.Annotated)((*Copy)(nil))
+
+// Copy implements Annotate which allows you to attach descriptions to the Copy resource.
+func (c *Copy) Annotate(a infer.Annotator) {
+	a.Describe(&c, "Copy a local file to a remote host.")
+}
+
+type CopyInputs struct {
+	Connection   *Connection      `pulumi:"connection" provider:"secret"`
+	Triggers     *[]interface{}   `pulumi:"triggers,optional" providers:"replaceOnDelete"`
+	LocalAsset   *asset.Asset     `pulumi:"localAsset,optional"`
+	LocalArchive *archive.Archive `pulumi:"localArchive,optional"`
```

Comment:
Any particular reason? I prefer the added clarity at the cost of verbosity.

Reply:
It's related to https://github.com/pulumi/pulumi-command/pull/423#discussion_r1597285198. Changing these names is breaking, and I don't think we want to always require a local asset to be used.

When users use a remote asset (say in typescript), then they would need to pass 

```ts
  localAsset: new pulumi.asset.RemoteAsset(remoteURL),
```

Since we will want to expand the `LocalAsset` to accept all kinds of Pulumi assets before v2, I would rather users write:

```ts
  asset: new pulumi.asset.RemoteAsset(remoteURL),
```

Reply:
Makes sense. I'm just a little concerned about clarity regarding the "from" and "to" of the copy. The "to" is always remote, but the "from" might also be when we open up to all kinds of assets. But `FromAsset` is not a great name either.